packages/odyssey-react/src/components/PasswordInput/PasswordInput.tsx

```diff
@@ -0,0 +1,81 @@
+/*!
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import React, { forwardRef, useState } from "react";
+import { withTheme } from "@okta/odyssey-react-theme";
+import { useOmit } from "../../utils";
+import { theme } from "./PasswordInput.theme";
+import styles from "./PasswordInput.module.scss";
+import { TextInput, TextInputProps } from "../TextInput";
+import { EyeIcon, EyeOffIcon } from "../Icon";
+import { Button } from "../Button";
+import { Tooltip } from "../Tooltip";
+import type { TooltipProps } from "../Tooltip";
+
+interface PasswordInputProps
+  extends Omit<TextInputProps, "type" | "prefix" | "suffix"> {
+  type?: never;
+  prefix?: never;
+  suffix?: never;
+  tooltipLabel?: TooltipProps["label"];
+  tooltipPosition?: TooltipProps["position"];
+}
+
+/**
+ * Password inputs exposes hide/show capability
+ */
+export const PasswordInput = withTheme(
+  theme,
+  styles
+)(
+  forwardRef<HTMLInputElement, PasswordInputProps>((props, ref) => {
+    const { name, label, tooltipLabel, tooltipPosition, ...rest } = props;
+
+    const omitProps = useOmit(rest);
+    const [internalType, setInternalType] = useState<"password" | "text">(
+      "password"
+    );
+
+    const onHideShowPasswordClick = () => {
+      setInternalType(internalType === "password" ? "text" : "password");
+    };
+
+    return (
+      <TextInput
+        {...omitProps}
+        ref={ref}
+        name={name}
+        type={internalType}
+        label={label}
+        suffix={
+          <span className={styles.affixFull}>
+            <Tooltip
+              label={tooltipLabel || label}
+              position={tooltipPosition || "bottom"}
+            >
+              <Button
+                name={name}
+                variant="affix"
+                icon={
+                  internalType === "password" ? <EyeIcon /> : <EyeOffIcon />
+                }
+                onClick={onHideShowPasswordClick}
+              />
```

Comment:
we'll need a new prop for the label contents for screen readers that will fix the a11y test error above

Reply:
should there be a `buttonLabel` prop? Should it be optional and default to label like the `tooltipLabel` does?
Or should I change all these labels to be required?

Reply:
it can't be optional because the interaction is behavior a non-sighted user will need to understand so we'll need to make it required!